src/coreclr/tools/Common/Compiler/InstructionSetSupport.cs

```diff
@@ -70,51 +70,33 @@ public static string GetHardwareIntrinsicId(TargetArchitecture architecture, Typ
             if (!potentialTypeDesc.IsIntrinsic || !(potentialTypeDesc is MetadataType potentialType))
                 return "";
 
+            // 64-bit ISA variants are not included in the mapping dictionary, so we use the containing type instead
+            if ((architecture, potentialType.Name) is (TargetArchitecture.X64, "X64") or (TargetArchitecture.ARM64, "Arm64"))
+                potentialType = (MetadataType)potentialType.ContainingType;
+
+            // We assume that managed names in InstructionSetDesc.txt use an underscore separator for nested classes
             string suffix = "";
-            if (architecture == TargetArchitecture.X64)
+            while (potentialType.ContainingType is MetadataType containingType)
             {
-                if (potentialType.Name == "X64")
-                    potentialType = (MetadataType)potentialType.ContainingType;
-                if (potentialType.Name == "VL")
```

Comment:
This handling of `VL` classes assumed the `IsSupported` value would be the same as the containing class.  Although that assumption is safe, it was unnecessary since they have their own mappings in the dictionary and can be handled by the generic nesting logic.

The special handling of the 64-bit classes had to stay.